guard/tests/validate.rs

```diff
@@ -6,672 +6,341 @@ pub(crate) mod utils;
     use cfn_guard::commands::validate::Validate;
-    use cfn_guard::commands::{DATA, INPUT_PARAMETERS, RULES, VALIDATE, SHOW_SUMMARY};
-    use cfn_guard::utils::writer::{Writer, WriteBuffer::Stdout, WriteBuffer::Vec as WBVec};
-    use indoc::indoc;
+    use cfn_guard::commands::{
+        ALPHABETICAL, DATA, INPUT_PARAMETERS, LAST_MODIFIED, OUTPUT_FORMAT, PAYLOAD,
+        PREVIOUS_ENGINE, PRINT_JSON, RULES, SHOW_CLAUSE_FAILURES, SHOW_SUMMARY, VALIDATE, VERBOSE,
+    };
+    use cfn_guard::utils::writer::{WriteBuffer::Stdout, WriteBuffer::Vec as WBVec, Writer};
+
+    use crate::utils::{
+        compare_write_buffer_with_file, compare_write_buffer_with_string,
+        get_full_path_for_resource_file, CommandTestRunner,
+    };
     use crate::{assert_output_from_file_eq, assert_output_from_str_eq};
-    use crate::utils::{get_full_path_for_resource_file, cfn_guard_test_command};
-
 
-    #[test]
-    fn test_single_data_file_single_rules_file_compliant() {
-        let data_arg = get_full_path_for_resource_file(
-            "resources/validate/data-dir/s3-public-read-prohibited-template-compliant.yaml",
-        );
-        let rules_arg = get_full_path_for_resource_file(
-            "resources/validate/rules-dir/s3_bucket_public_read_prohibited.guard",
-        );
-
-        let data_option = format!("-{}", DATA.1);
-        let rules_option = format!("-{}", RULES.1);
-        let args = vec![VALIDATE, &data_option, &data_arg, &rules_option, &rules_arg];
-
-        let mut writer = Writer::new(Stdout(stdout()));
-        let status_code = cfn_guard_test_command(args, &mut writer);
-        assert_eq!(0, status_code);
+    #[derive(Default)]
+    struct ValidateTestRunner<'args> {
+        data: Vec<&'args str>,
+        rules: Vec<&'args str>,
+        show_summary: Vec<&'args str>,
+        input_parameters: Vec<&'args str>,
+        output_format: Option<&'args str>,
+        previous_engine: bool,
+        show_clause_failures: bool,
+        alphabetical: bool,
+        last_modified: bool,
+        verbose: bool,
+        print_json: bool,
+        payload: Option<&'args str>,
     }
 
-    #[test]
-    fn test_single_data_file_single_rules_file_compliant_verbose() {
-        let data_arg = get_full_path_for_resource_file(
-            "resources/validate/data-dir/s3-public-read-prohibited-template-compliant.yaml",
-        );
-        let rules_arg = get_full_path_for_resource_file(
-            "resources/validate/rules-dir/s3_bucket_public_read_prohibited.guard",
-        );
-        let show_summary_arg = "all";
-
-        let data_option = format!("-{}", DATA.1);
-        let rules_option = format!("-{}", RULES.1);
-        let show_summary_option = format!("-{}", SHOW_SUMMARY.1);
-        let args = vec![VALIDATE,
-                        &data_option, &data_arg,
-                        &rules_option, &rules_arg,
-                        &show_summary_option, &show_summary_arg];
+    impl<'args> ValidateTestRunner<'args> {
```

Comment:
Wondering if we should reuse this struct in actual execution of the command too?